nap/utils.py

```diff
@@ -1,9 +1,36 @@
 from __future__ import unicode_literals
+from collections import namedtuple
 import itertools
+from urlparse import urlparse
+
 import six
 from six.moves import urllib
 
 
+def normalize_url(url_string):
+    QueryParam = namedtuple("QueryParam", "key value")
+    url = urlparse(url_string)
+    if not url.query:
+        return url_string
+
+    query_params = []
+    for entry in url.query.split('&'):
+        key_value = entry.split("=")
+        query_params.append(QueryParam(key=key_value[0], value=key_value[1] if len(key_value) == 2 else None))
+    query_params.sort()
+
+    query_normalized = "?"
+    for index, param in enumerate(query_params):
+        if index > 0:
+            query_normalized += '&'
+
+        query_normalized += param.key
+        if param.value:
+            query_normalized += "={}".format(param.value)
+
+    return url.scheme + "://" + url.netloc + url.path + url.params + query_normalized + url.fragment
```

Comment:
`urlunsplit` (or `urlunparse`) would nicely mirror the splitting/parsing at the beginning of this function, and could potentially save you from subtle security vulnerabilities (and probably support more diverse URL variations).

https://docs.python.org/2/library/urlparse.html#urlparse.urlunsplit